### PR TITLE
wallet: migrate mirage from js to ts

### PR DESCRIPTION
**Pull request by maxime-carabina**
We are updating `mirage` from `JS` to `TS`.

### CODE REVIEW

File: web-frontend/src/mirage/factories/account.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    return faker.internet.userName();
+  },
+  candidateBalance() {
+    return faker.number.int().toString();
+  },
+  balance() {
+    return faker.number.int().toString();
+  },
+  address() {
+    return 'AU' + faker.string.alpha({ length: { min: 128, max: 256 } });
```
Comment (Pierregvx) [inline web-frontend/src/mirage/factories/account.ts:16]: not blocking but if I am not wrong an address is 53 char max

File: web-frontend/src/mirage/factories/account.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  keyPair() {
+    return {
+      privateKey: faker.string.alpha({ length: { min: 128, max: 256 } }),
+      publicKey: faker.string.alpha({ length: { min: 128, max: 256 } }),
+      nonce: faker.lorem.word(),
```
Comment (Pierregvx) [inline web-frontend/src/mirage/factories/account.ts:22]: isn't nonce a number ?

**Review by Pierregvx — APPROVED** [review]

File: web-frontend/src/mirage/factories/account.ts (hunk 2 above)
Comment (Pierregvx) [reply web-frontend/src/mirage/factories/account.ts:22]: in case we use it in the front I think we should set it as a number.toString()

Comment (maxime-carabina) [reply web-frontend/src/mirage/factories/account.ts:22]: I just copy the same factory as in the past. But it's a good question 🤔

File: web-frontend/src/mirage/factories/account.ts (hunk 1 above)
Comment (maxime-carabina) [reply web-frontend/src/mirage/factories/account.ts:16]: No! Address can be long but we can check that with @Thykof  🤔

Comment (Thykof) [reply web-frontend/src/mirage/factories/account.ts:16]: for the sake of the faker, min 50 max 53

File: web-frontend/src/mirage/factories/account.ts (hunk 2 above)
Comment (maxime-carabina) [reply web-frontend/src/mirage/factories/account.ts:22]: > see here
> 
> https://github.com/massalabs/station-massa-wallet/blob/8507322fb6d9ea85d6a1f0fbe15d060163a94cc7/api/walletApi-V0.yml#L95
> 
> it depends on the ciphered query param: sensitive data (private key, nonce, salt) are returned only if ciphered if false. It's not words, it's base58CheckEncoded binary data: https://github.com/massalabs/station-massa-wallet/blob/6ca35c3040b3746ec56fd6fb0e714abdae61f76c/pkg/wallet/wallet.go#L529C44

So.... what can I do with faker ?

Comment (Thykof) [reply web-frontend/src/mirage/factories/account.ts:22]: empty strings ^^ 

It depends what we want to simulate: backup private key? then we need a fake private key in the body response

Comment (maxime-carabina) [reply web-frontend/src/mirage/factories/account.ts:22]: > empty strings ^^
> 
> It depends what we want to simulate: backup private key? then we need a fake private key in the body response

Ok! I will put an empty string for now 😉

Comment (Thykof) [reply web-frontend/src/mirage/factories/account.ts]: to reflect the API all sensitive data (private key, nonce, salt) should be empty string or random string

**Review by Thykof — APPROVED** [review]

**Pull request merged**